references/validationFunctions/specimen/percentageTumourCells.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of the GNU Affero General Public License v3.0.
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ */
+
+const validation = () =>
+(function validate(inputs) {
+    // When $name is percent_tumour_cells, and $field is a number between 0 and 1
+    const {$row, $name, $field} = inputs;
+
+    const result = { valid: true, message: 'Ok' };
+    const measurementMethodExceptionTypes = ['not applicable'];
+
+    // checks for a string just consisting of whitespace
+    const checkforEmpty = (entry) => {
+      return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
+    };
+
+    const fieldHasValue = $field && $field != null && !(checkforEmpty($field));
+    const measurementMethod = $row?.percent_tumour_cells_measurement_method?.trim?.().toLowerCase();
+
+    if (fieldHasValue) {
+      if (measurementMethodExceptionTypes.includes(measurementMethod)) {
```

Comment:
if `percent_tumour_cell` has a value AND  `percent_tumour_cell_measurement_method` is "not applicable" (exceptions list is extendable here to "unknown" etc.), then validation fails